POEApi.Model/Stash.cs

```diff
@@ -64,6 +65,34 @@ public void RefreshAll(POEModel currentModel, string currentLeague, string accou
             }
         }
 
+        public void RefreshUsedTabs(POEModel currentModel, string currentLeague, string accountName)
+        {
+            var usedTabIds = items.GroupBy(i => i.InventoryId)
+                .Select(group => group.First())
+                .Select(item => item.InventoryId)
+                .Where(id => !string.IsNullOrWhiteSpace(id))  // An UnknownItem could have an invalid InventoryId.
+                .Select(id => Int32.Parse(Regex.Match(id, @"\d+").Value) - 1);
```

Comment:
Difficult to read and digest what this linq is doing. Regex.Match(bla) may throw a null ref for the .Value.

Reply:
The real annoyance with LINQ expressions is that they are fantastic for doing something complicated in a really compact way, but they are both obtuse and do not lend themselves to error checking.  They look great on the outside, but they're not good for you.  The candy of programming patterns.

Reply:
Agreed.